### ISSUE TITLE
JWKS endpoint returns x5t#s256 value incompatible with the RFC specification

### ISSUE THREAD

**Issue by pawelwso2is** [wso2/product-is#14899]
**Describe the issue:**
Using WSO2IS 6.0.0. When calling oauth2/jwks endpoint it returns some new fields that were not available in the previous versions. In particular x5t#s256 is returned which seems to be double base64url encoded hex-string value of the fingerprint:
```
{
  "keys": [
    {
      "kty": "RSA",
      "x5t#S256": "TkRSaU1UTTNOR1kzTURSaE56QmpaamMwTTJJNFpqVTROV015TURsaU1XRTRNamN3WVRobVltUTRZalExTXpVeVpURmhOMlptWWpVek1qSTJOakZqTnc",
      "e": "AQAB",
      "use": "sig",
      "kid": "NDRiMTM3NGY3MDRhNzBjZjc0M2I4ZjU4NWMyMDliMWE4MjcwYThmYmQ4YjQ1MzUyZTFhN2ZmYjUzMjI2NjFjNw_RS256",
      "x5c": [.....
.....
```
when base64url decode the value of the x5t#256 field, I got the value equal to kid+_RS256 and when bas64url decode it once again, I've got something like
44b1374f704a70cf743b8f585c209b1a8270a8fbd8b45352e1a7ffb5322661c7
what is string representation of the sha256 fingerprint. (https://base64.guru/standards/base64url/decode)
So the value of the x5t#S265 field doesn't comply with the RFC  (https://www.rfc-editor.org/rfc/rfc7515#page-12):

**The "x5t#S256" (X.509 certificate SHA-256 thumbprint) Header
   Parameter is a base64url-encoded SHA-256 thumbprint (a.k.a. digest)
   of the DER encoding of the X.509 certificate [RFC5280] corresponding
   to the key used to digitally sign the JWS.**

Based on the specification it should be 32 bytes base64url encoded and obviously the value returned in the json cause oauth2 libraries failing when trying to validate the token with jwks endpoint.

**How to reproduce:**
1. query JWKS endpoint 
`curl -k --user admin:admin -X GET  --header "Content-Type:application/json" 'https://wso2is.server:9443/oauth2/jwks' | jq -r '.'
`

**Expected behavior:**
xt5#s256 returned value as per RFC specification

**Environment information:**
 - Product Version: IS 6.0.0
 - OS: Linux
 - Database: Postgres
 - Userstore: LDAP

---

**Comment by senthalan** [comment]
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2067

**Comment by Dinithi-Hasanika** [comment]
This issue needs to be fixed in Identity server 5.10

**Comment by ThaminduDilshan** [comment]
Known issue https://github.com/wso2/product-is/issues/18817
